src/AsmResolver.DotNet/ReferenceImporter.cs

```diff
@@ -332,6 +367,71 @@ public virtual TypeSignature ImportTypeSignature(Type type)
             return new TypeDefOrRefSignature(reference, type.IsValueType);
         }
 
+        // separate class to prevent static cctor checks from leaking into the other ReferenceImporter methods on NAOT
+        private static class FnPtrHelper
+        {
```

Comment:
This class seems somewhat weirdly positioned in between all other methods. We may want to move it to a separate file solely related to these types of helpers. We can then also move the `Marshal.GetHINSTANCE` reflection hack in [ModuleDefinition::FromModule](https://github.com/Washi1337/AsmResolver/blob/6f0a38370372ffc8029a5c1409ea6611bdd4c6a8/src/AsmResolver.DotNet/ModuleDefinition.cs#L171). Besides, I also expect we will get more reflection "hacks" like this in the future.